src/Microsoft.AspNetCore.Routing/Matchers/DefaultEndpointSelector.cs

```diff
@@ -11,17 +11,34 @@ namespace Microsoft.AspNetCore.Routing.Matchers
 {
     internal class DefaultEndpointSelector : EndpointSelector
     {
+        private readonly IEndpointSelectorPolicy[] _policies;
+        
+        public DefaultEndpointSelector(IEnumerable<MatcherPolicy> policies)
+        {
+            if (policies == null)
+            {
+                throw new ArgumentNullException(nameof(policies));
+            }
+
+            _policies = policies.OrderBy(p => p.Order).OfType<IEndpointSelectorPolicy>().ToArray();
```

Comment:
My first impression is it feels a little weird that something has to inherit from `MatcherPolicy` and implement `IEndpointSelectorPolicy`.

Reply:
Them both ending in Policy is also a little confusing.

Reply:
And `policies` + `_policies`. Do you think Policy is being overloaded?

Reply:
Well, the reason why it's separated this way is that there's a bunch of policies that all play together. Maybe it would make sense as a marker interface that they all inherit from as well....

I'll log an issue for this.

Reply:
I'll rename these variables for now. If we want to have a discussion about the naming of extensibility points lets do that during API review.